tmt/steps/discover/__init__.py

```diff
@@ -192,7 +231,33 @@ def base_command(cls, method_class=None, usage=None):
         @click.option(
             '-h', '--how', metavar='METHOD',
             help='Use specified method to discover tests.')
+        @click.option(
+            '--fmf-id', default=False, is_flag=True,
+            help='Show fmf identifiers for tests discovered in plan.')
         def discover(context, **kwargs):
+            if kwargs.get('fmf_id'):
+                if kwargs.get('how') == 'shell':
+                    raise tmt.utils.GeneralError(
+                        f"`tmt run discover --fmf-id` is supported only"
+                        f" for `fmf` method.")
+                import subprocess
+                context.parent.params['quiet'] = True
+                context.parent.params['debug'] = 0
+                context.parent.params['verbose'] = 0
+                # Raise an exception if --fmf-id uses w/o --url and git root
+                # doesn't exist for ALL discovered plans
+                url = tmt.utils.Common.get_fmf_attr('discover', 'url')
```

Comment:
this shouldn't be necessary when run in `DiscoverFmf`'s go() - it should take care merging cmdline/plan options (opt() call)

Reply:
I've tried to move that to `DiscoverFmf.go()` and it works fine in all cases except one:
"fmf-id (w/o and w/ url): non-git directory": If in non-git directory exists plan w/ and w/o url, then only fmf-ids for the plan w/ url should be reported.
Without `get_fmf_attr` the exception will be raised when non-url plan will be found and the whole output will be mess. It will look like the list of fmf-ids with `tmt.utils.GeneralError` at the bottom. Or just `GeneralError` at the top.
I can still move the whole piece of code to `DiscoverFmf.go()` but **with** `get_fmf_attr`

Reply:
UPD: I've decided to move the code above **with** `get_fmf_attr` to `DiscoverFmf.go()`. It makes sense. All tests passed except fedora-rawhide: there is an infrastructure issue.